validator/keymanager/v2/derived/derived.go

```diff
@@ -406,27 +402,38 @@ func (dr *Keymanager) DepositDataForAccount(accountIndex uint64) ([]byte, error)
 	return tx.Data(), nil
 }
 
-func (dr *Keymanager) initializeSecretKeysCache() error {
-	dr.lock.Lock()
-	defer dr.lock.Unlock()
-	for i := uint64(0); i < dr.seedCfg.NextAccount; i++ {
+// Append public and secret keys for a given account index to their respective caches
+func (dr *Keymanager) appendKeyesToCaches(secretKey bls.SecretKey) error {
+	publicKey := bytesutil.ToBytes48(secretKey.PublicKey().Marshal())
+	dr.publicKeysCache = append(dr.publicKeysCache, publicKey)
+	dr.secretKeysCache[publicKey] = secretKey
+	return nil
+}
+
+// Initialize public and secret key caches used to speed up the functions
+// FetchValidatingPublicKeys and Sign
+func (dr *Keymanager) initializeKeysCaches() error {
+	count := dr.seedCfg.NextAccount
```

Comment:
It definitely feels like this function should be the one to hold to the lock

Reply:
As mentioned in the above comment, it is only called during the initialization of the keymanager, there is no real need to protect it from race conditions.